parcels/kernel.py

```diff
@@ -234,8 +234,8 @@ def load_lib(self):
 
     def execute_jit(self, pset, endtime, dt):
         """Invokes JIT engine to perform the core update loop"""
-        if len(pset) > 0 and pset.particle_data['xi'].ndim == 2 and pset.fieldset is not None:
-            assert pset.fieldset.gridset.size == pset.particle_data['xi'].shape[1], \
+        if len(pset) > 0 and pset.collection._data['xi'].ndim == 2 and pset.fieldset is not None:
+            assert pset.fieldset.gridset.size == pset.collection._data['xi'].shape[1], \
```

Comment:
just as I saw before, if you actually wanna change the state of the 'data' of a collection, a proper setter-getter would be appropriate for the data array, otherwise we're just breaking the member variable protection 'convention'

Reply:
Absolutely, every line of code that looks like this means that section needs to be reworked. The purpose of changing these lines in this way, is to make them work again (for now at least). That is step 1. Step 2 is to gradually replace all code that depends on the specific implementation of the datastructure by code that does not, like Raoul has been doing in the execute function below.

Reply:
@CKehl For this part of the code I made a particle_field_check() as a member of the particle set. Does that make sense to you? 

Is there a reason why this check is performed in execute_jit(), but not in execute_python()?

Reply:
I can answer that question. It's a legacy choice because we used to only have particle.xi in JIT mode. However, since #940 we also support particle.xi in Scipy mode, so I guess that this check also needs to be done in `execute_python()`